dev/build.example.testcontainers_fat/test-applications/app/src/web/dbrotation/DbRotationServlet.java

```diff
@@ -80,4 +84,28 @@ public void testDatabaseRotationWithNoAuth() throws Exception {
             //No auth data was provided, therefore expect an error!
         }
     }
+
+    /**
+     * Ensure the general use case of using ojdbc8.jar with Java 17/21 works.
+     *
+     * Oracle's JDBC driver is more tightly coupled to JDK classes than
+     * other JDBC drivers so we want to make sure forward compatibility is achievable.
+     */
+    @Test
+    @OnlyIfSysProp(DB_Oracle)
+    @MinimumJavaLevel(javaLevel = 17)
+    public void testOracleJava17() throws Exception {
```

Comment:
Add `@MaximumJavaLevel(javaLevel = 20)` here to prevent this test being run twice at Java 21+?